### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by orubel**
Resilience needs to be done through 'syncronization' not 'replication'; replication suggests duplicitous data creating and a cross cutting concern. This is against AOP principles. Within a database system, replication is equivalent to syncronization but the difference being that syncronization as a KEYWORD is significant in the fact that syncronized data is NOT replicated and/or duplicated; it is a representation of the same data across all systems.

If one must use the term 'replicated', one should use it in conjunction with 'syncronized' to be 100% clear.

### CODE REVIEW

**Comment by lazaridis-com** [comment]
I'll not discuss if @orubel 's remarks are valid or not.

Just a general comment, subjecting the relevant part:

> Resilience is achieved by 'syncronization', containment, [isolation](/glossary#Isolation) and [delegation](/glossary#Delegation).

should be removed from the document, as it affects the **implementation** of a **trait** (Resilience), suggesting that the mentioned means (synchronization, containment, isolating and delegation) are the exact (and only) combination that can achieve resiliency, which is not true.

**Comment by zhenglaizhang** [comment]
@lazaridis-com Agreed with you, the manifesto should generally points out principles, describing what react programming is at the high level, it may not point out the implementation.

**Comment by orubel** [comment]
'replication' is not 'synchronization'. Synchronization does not require replication. If anything, ADD the word 'synchronization' to documentation

**Comment by orubel** [comment]
For example, replication can cause a cross cutting concern because data contained in functionality is not synchronized. As an example, API's in a distributed architecture create an architectural cross cutting concern due to replication without synchronization.

This leads to the following issues:  slideshare.net/bobdobbes/university-of-washington-the-new-api-pattern#14

**Comment by rkuhn** [comment]
@orubel Synchronization as a primary tool runs counter to the stated goal of this manifesto: it is the realization that synchronicity is impossible for distributed agents—this is a simple consequence from Einstein’s physics. Building synchronous(*), distributed systems led to the failures of previous approaches, to the impossibility of infinite scaling, to the inability of tolerating failures without denying service. The resulting loss of consistency is inherent to the nature of distribution itself, as has been proven in multiple ways (CAP, FLP, … ).

@lazaridis-com In the light of the paragraph above, it is the duty of the manifesto to spell out those measures that differentiate reactive systems from approaches that do not scale robustly to an arbitrary degree. Removing these points makes “reactive” the void phrase that some people want it to be, a pure marketing term. This manifesto puts a stake in the ground precisely to give concrete meaning and a technically useful definition, without being prescriptive on the implementation details.

In summary, I don’t see an argument being made here that would convince me of a change.

---

(*) There are notions of synchronicity that are not usually meant in this context, see for exactly Virtual Synchrony, that I don’t think anyone of us is currently discussing.

**Comment by lazaridis-com** [comment]
@rkuhn , my arguments are within #48 and other issues I've filed around two years ago, after thinking deeply about all those issue for days, possibly weeks.

The "duty" of the Manifesto? Specify it first: #53 - And then you'll see that "Manifesto" is the description/definition/specification of a general-validity concept.

If you want to keep it as-is, just rename it to "The Lightbend Manifesto", and its fine (I mean this, that's not an ironic statement).

(btw: you can close this PR here, because you won't merge it. Would anyway not make sense to change the one implementation detail with another one)

**Comment by rkuhn** [comment]
@lazaridis-com I’m not affiliated with Lightbend.

**Comment by lazaridis-com** [comment]
@rkuhn , this is not relevant. You can even better rename it to "The Akka Manifesto".

(i'm anyway out here, as I'll not use Java/Scala for my personal projects, so its not much of relevance for me anymore. The essence is within #48, it's even a lingustic matter, so it possibly will take you 5 to 10 years to get to it. Again, no irony here.)

**Comment by orubel** [comment]
@rkuhn Synchronization is not impossible. To state so means you just dont know how. I have given talks to Netflix, Apple, VMWare, Amazon and others on synchronized IO state. It is extremely possible and very doable. I have multiple talks, videos and conference slides on it and have spoken at the U of CA -Berkeley and recently U of WA on this topic.

This issue lies with binding communication logic with the business logic creating a secondary binding and an architectural cross cutting concern. Abstract the communication layer from the business logic and you are able to solve for the architectural cross cutting concern and synchronize the I/O state

https://www.slideshare.net/bobdobbes/university-of-washington-the-new-api-pattern

**Comment by rkuhn** [comment]
Of course it is possible to synchronize multiple agents, but then we are no longer talking about a distributed system because partial failure is no longer permitted. Such a setup does not increase reliability or provide resilience, it instead multiplies the probability of failure.

**Comment by orubel** [comment]
Synchronized I/O state improves reliability of the entire system. So are you saying you'd rather have separate system fall into unsynchronized states vs being in synchronized states? You do realize this causes an architectural cross cutting concern and issues with the following : https://www.slideshare.net/bobdobbes/university-of-washington-the-new-api-pattern#14

**Comment by orubel** [comment]
I am shocked and in awe that people are embracing AOP here but dont understand what a cross cutting concern is in the architecture... Especially when it comes to I/O. Thats what this entire discussion seems to come down to. 

If you replicate you I/O data (not function), you risk your machines falling escalating privileges.

This is bad.

You want that I/O state to ALWAYS be in sync. 

And in the since of replicating function, you want it to be syncronized and pulled from a common repository.

Again, the verbage of syncronization is VERY important.

You dont just hapharzardly replicate across your architecture... you syncronize your data and your functionality... unless you are deliberately wanting things to be DIFFERENT.

KISS,DRY.

**Comment by orubel** [comment]
@rkuhn The two things I am hearing from you are:

- Synchronization of I/O is " impossible for distributed agents" 
- Once you synchronize I/O, it is no longer a distributed system (ie ' it is possible to synchronize multiple agents, but then we are no longer talking about a distributed system because partial failure is no longer permitted').

Is this correct? Because I was thinking of writing an article on this conversation and the fact that the maintainers of the Reactive Manifesto are making these claims.

**Comment by rkuhn** [comment]
@orubel We are not talking about the same things, and I am certain that we are not using the same language. Here are a few examples:

* “I/O data” does not make sense to me, unless you mean the bytes in a TCP stream
* “I/O state” could possibly be the state of a TCP socket in the kernel, but that seems unrelated
* “synchronize I/O” does not make sense to me, what would that mean?

Another point that I cannot interpret is “people are embracing AOP here”, how do you come to this conclusion?

---

Independent of the above, a distributed system is one where agents are spread across multiple locations and communicate only by message passing. These systems experience partial failure, which distinguishes them from local but concurrent systems. When using a synchronization protocol, for example a consensus algorithm, partial failure is eliminated at the cost of failing when enough agents fail, hence building a non-distributed system on top of distributed low-level agents. This implies multiplied susceptibility to failure, i.e. it decreases reliability compared to using a single agent.

**Comment by orubel** [comment]
**“I/O data” does not make sense to me, unless you mean the bytes in a TCP stream**
ok... assume a distributed architecture of proxy/api server/MQ. The request/response (I/O) has to be able to redirect and forward from anyone of those machines back into any of the others (effectively).

To do that, we must share and sync the I/O data ... all data related to the request/response for what is being accessed (ie endpoint). Thus when the request hits the proxy, it has the same data as the rest of the architecture and upon a redirect from api server or MQ, they too all have the exact same data regardless of what is being requested. All I/O data is in SYNC across the architecture. 

Some people do this at an 'api gateway' but because there is a binding of I/O data at the business logic, this data can get out of sync merely due to human error and causes ELEVATED PRIVILEGES.

Also redirects cannot assume to be handled the same way from one service to the next when none are synchronized from a central source (ie like a repo).

**“I/O state” could possibly be the state of a TCP socket in the kernel, but that seems unrelated

I/O state is all data related to communication for requested resource. in AOP, we learn to separate process by what they are focused on. I/O is not focused on fetching or manipulation of resource; it is only focused on communication and fulfilling request/response. Doesn't care what it fulfills it WITH.

**“synchronize I/O” does not make sense to me, what would that mean?****

One would use one file to synchronize the state of I/O across all systems. So when a request is made, the proxy has the exact same information as the api server and as the MQ. If you have seen RAML, API Blueprint, you are similar with what it would be like.

OF COURSE... this is just one principle of synchronizing in your app to maintain good development principles. You do this all the time in:

- devops with Puppet, Chef, Ansible, etc
- repositories (Artifactory, Maven)

Synchronization is a KEY PRINCIPLE... not duplication. Duplication MAY OCCUR but a centralized system always makes sure things stay in sync.

**Comment by rkuhn** [comment]
The concern will be addressed in #100.
